core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -115,26 +130,19 @@ object Validator {
           )
       }
     }
-  }
+  }.toZIO
 
   private def check(
     document: Document,
     rootType: RootType,
     variables: Map[String, InputValue]
-  ): IO[ValidationError, Map[String, FragmentDefinition]] = {
+  ): EReader[Any, ValidationError, Map[String, FragmentDefinition]] = ZPure.suspend {
```

Comment:
Same as previously, I think the `ZPure.suspend` might not be needed

Reply:
Removed